tsdb/engine/wal/wal.go

```diff
@@ -0,0 +1,1591 @@
+/*
+Package WAL implements a write ahead log optimized for write throughput
+that can be put in front of the database index.
+
+The WAL is broken into different partitions. The default number of
+partitions is 5. Each partition consists of a number of segment files.
+By default these files will get up to 2MB in size before a new segment
+file is opened. The files are numbered and start at 1. The number
```

Comment:
just because. 0 indexing is for arrays. If you're counting things (like files) you generally start at 1